packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1360,3 +1361,103 @@ class KeepAlive extends ParentDataWidget<KeepAliveParentDataMixin> {
     properties.add(DiagnosticsProperty<bool>('keepAlive', keepAlive));
   }
 }
+
+// Return a Widget for the given Exception
+Widget _createErrorWidget(Object exception, StackTrace stackTrace) {
+  final FlutterErrorDetails details = FlutterErrorDetails(
+    exception: exception,
+    stack: stackTrace,
+    library: 'widgets library',
+    context: ErrorDescription('building'),
+  );
+  FlutterError.reportError(details);
+  return ErrorWidget.builder(details);
+}
+
+/// Set a cross axis extent constraint to the child Sliver. By default, slivers will try to
+/// fill up the available cross axis extent, but this parent data widget will override that
+/// behavior.
+class SliverConstrainedCrossAxis extends ParentDataWidget<SliverFlexParentData> {
+  /// Constructor.
+  const SliverConstrainedCrossAxis({
+    super.key,
+    required this.extent,
+    required Widget sliver,
+  }): super(child: sliver);
```

Comment:
Should there be some assertions? Can extent be double.infinity? Can it be <0?

Reply:
Added an assertion to ensure 0 < extent < double.infinity and same for flex as well.